### PR TITLE
AVX-58131: update terraform documentation for external groups

### PR DESCRIPTION
**Pull request by kasunkarunaratne**

### CODE REVIEW

File: docs/resources/aviatrix_smart_group.md [hunk 1]
```diff
@@ -56,13 +56,43 @@ resource "aviatrix_smart_group" "test_smart_group_ip" {
...
+      }
+    }
+
     match_expressions {
-      external = "External_ID"
+      external = "threatiq"
       ext_args = {
-        external_ID_specific_field1 = "value1"
-        external_ID_specific_field2 = "value2"
-      }
```
Comment (shinrich) [inline docs/resources/aviatrix_smart_group.md:64]: Do you want to keep the generic example too?

Comment (kasunkarunaratne) [reply docs/resources/aviatrix_smart_group.md:64]: added back, thanks

**Review by shinrich — APPROVED** [review]

**Comment by Dennizz** [comment]
- Do we have documented somewhere what ext_args are available? Are these examples exhaustive?
- In the external field, it states _currently either "geo" or "threatiq"_. In the examples you also have other sources. Are the external sources in the examples exhaustive?
- How can users learn about all available fields in a certain external source?

**Review by dnardo — APPROVED** [review]

**Comment by kasunkarunaratne** [comment]
> * Do we have documented somewhere what ext_args are available? Are these examples exhaustive?
> * In the external field, it states _currently either "geo" or "threatiq"_. In the examples you also have other sources. Are the external sources in the examples exhaustive?
> * How can users learn about all available fields in a certain external source?

- The list of available ext_args are returned by an api , https://github.com/AviatrixDev/cloudn/blob/40953da9b426093a4ae6dbff984af35ee725d619/cloudx-local/daemon/v2_5/schemas/external_groups.yaml#L34. The examples are exhaustive up to 7.2.b but theoretically these fields can change depending on the downloaded dbs.
- 7.2.b (update - copilot 4.17) documentation should include "azureips" and "githubips" as well. Yes, source types are exhaustive up to 7.2.b.
- Using the `resolve` api mentioned above

**Pull request merged**